docs/reference/dag.md

```diff
@@ -3,6 +3,53 @@ id: dag
 title: DAG
 ---
 
-:::warning
-This page is under review and currently not visible in the menu.
-:::
+The Directed Acyclic Graph (DAG) describes the data pipeline and the dependencies between actions. 
+It is build automatically within SDLB by analyzing the actions, related dataObjects and its dependencies. 
+SDLB uses this DAG to optimize execution of the pipeline as it knows in which order the pipeline needs to be executed.
+Because it's acyclic, it cannot have loops, so it's impossible to define an action that depends on an output of subsequent actions.
```

Comment:
@pgruetter Could you describe in more detail what you have in mind with recursiveInputs please?

Reply:
Yes, sure. You'll find the attribute here: https://smartdatalake.ch/json-schema-viewer/index.html#viewer-page&version=sdl-schema-2.3.0-SNAPSHOT.json&v=3-1-14
For special cases, we need to read the existing data of a DataObject and write it in the same action. You can do that in a custom action by defining a recursive input. 
But thinking about it again, maybe we first need to have separate article describing recursiveInput and then link to it as it may be a bit confusing just pointing to the schema browser.
Maybe just create a TODO as comment that we should link it once we have a description?

Reply:
Ahh sorry, sure the recusiveInputIds option, indeed provides an exception for the acyclic graph. I added a new page about actions and a note about recusiveInputIds in there and linked it. 
Now we have even more pages to fill ;)